src/repl-frontend/scala/tools/nsc/interpreter/shell/ShellConfig.scala

```diff
@@ -65,7 +65,14 @@ trait ShellConfig {
   // This property is used in TypeDebugging. Let's recycle it.
   val colorOk = Properties.coloredOutputEnabled
 
-  val historyFile = s"$userHome/.scala_history_jline3"
+  val datadir =
+    try {
+      val projectdirs = ProjectDirectories.from("org", "scala-lang", "repl2")
+      projectdirs.dataLocalDir
+    } catch {
+      case _: UnsupportedOperationException => userHome
```

Comment:
Under what circumstances do we anticipate `UnsupportedOperationException` potentially occurring?

Related question: what happens if directories-jvm is, due to some mishap, not present on the classpath? Does the behavior degrade gracefully?

Reply:
Apparently,
```
public class UnsupportedOperatingSystemException extends UnsupportedOperationException
```

Reply:
Is the lib not just making strings but also running shells to probe for dirs? I can't tell by looking, I just shifted back to an early morning schedule and won't be awake for another couple of hours.

Offhand, I would prefer not to spawn any (more) processes. Perhaps I was lazy in my due diligence.